vlib/v/README.md

```diff
@@ -0,0 +1,109 @@
+# Compiler pipeline
+A simple high level explanation
+how the compiler pipeline (`parser` -> `checker` -> `generator`) works.
+
+## Reading files
+### Getting builtin files
+To load all builtin files,
+a preference `Preferences.lookup_path` for the path where to look for exists.
+See `Builder.get_builtin_files` as example.
+If the file is a `.vsh` file and the backend is C, `vlib/os` will also be loaded as builtin.
+
+### Getting project files
+Either you there is a specific file: `my_file.v` or a directory containing V files.
+In the last case it scans that directory for all files.
+See `Builder.v_files_from_dir` as the helper method.
+This list of files needs to be filtered that only `*.v` files exist.
+
+Skips the following file types:
+- `*_test.v`
+- either `*.c.v` or `*.c.js` depending on the backend
+- all files that doesn't end with `.v`
+- Files that are not defined in `Preferences.compile_defines`
+or `Preferences.compile_defines_all` **if any file is defined**.
+
+## Parsing files
+To parse something a new template is created as the first step:
+```v
+import v.table
+table := table.new_table()
+```
+a new preference is created:
+```v
+import v.pref
+pref := pref.Preferences{}
+```
+and a new scope is created:
+```v
+import v.ast
+scope := ast.Scope{
+    parent: 0
+}
+```
+after that, you can parse your files.
+
+## Parse text
+If you want to parse only text which isn't saved on the disk you can use this function.
+```v oksyntax
+import v.parser
+code := ''
+// table, pref and scope needs to be passed as reference
+parsed_file := parser.parse_text(code, table, .parse_comments, &pref, &scope)
+```
+
+## Parse a single file
+For parsing files which are on disk it needs to be a path provided.
+The paths are collected one step earlier.
+```v oksyntax
+import v.parser
+path := ''
+// table, pref and scope needs to be passed as reference
+parsed_file := parser.parse_file(path, table, .parse_comments, &pref, &scope)
+```
+
+## Parse a set of files
+If you have a batch of paths available which should be parsed,
+there is also a function which does all the work.
+```v oksyntax
+import v.parser
+paths := ['']
+// table, pref and scope needs to be passed as reference
+parsed_files := parser.parse_files(paths, table, &pref, &scope)
+```
+
+## Parse imports
+A file often contains imports. This imports might needs to be parsed as well.
```

Comment:
`This imports might needs to be parsed as well.` -> `These imports might need to be parsed as well.`